source/extensions/filters/http/ext_proc/ext_proc.cc

```diff
@@ -477,7 +477,8 @@ FilterTrailersStatus Filter::onTrailers(ProcessorState& state, Http::HeaderMap&
     // We would like to process the body in a buffered way, but until now the complete
     // body has not arrived. With the arrival of trailers, we now know that the body
     // has arrived.
-    sendBufferedData(state, ProcessorState::CallbackState::BufferedBodyCallback, true);
+    sendBodyChunk(state, *state.bufferedData(), ProcessorState::CallbackState::BufferedBodyCallback,
+                  false);
```

Comment:
qq: you change here from `true` to `false`. Is this because that we could have trailers later or other body; or both?

Reply:
I am wondering why client (i.e., what is the use case) want to terminate the stream in such way? 

If client' doesn't terminate stream here, eventually the stream should still be destroy.

Reply:
Just want to confirm:
1) In normal request/response stream, an empty chunk data is sent from downStream to upStream. Also, `end_stream` is set in this case for this normal main stream?

2) Then when ext_proc body mode is set, ext_proc sends empty chunk to sideStream server. with `end_stream`, so it will terminate the side stream.

Reply:
For 1) correct, end_stream is set to true to indicate client does not have any further data to send.  Here "terminate" means no further data to send. It does not means shut down the connection or destroy the steam.

For 2) an empty chunk, like "",  is a valid data. If there is data come in, and if ext_proc mode is set to sent data, that chunk will be sent to ext_proc server. Server can mutate on it, as normal data, and send back the response to Envoy.   
Note, an empty chunk data is complete different from no data.  In the case of no data, no body is received by Envoy, and decodeData() is not called at all.

Reply:
Cool! Yes, we can keep this behavior for mutation from server.